### ISSUE TITLE
removed "Secret" setting prevents plugin boot

### ISSUE THREAD

**Issue by revolunet** [waseem18/mattermost-plugin-google-calendar#1]
Looks like this setting has been removed by plugin.json [here](https://github.com/waseem18/mattermost-plugin-google-calendar/commit/c097f2c123be3ac2bc9665c6ea71795ac0ecb545#diff-b3568fa08e53a00f66841ed9ca2c53d4)

But, on master branch, there is still [a check on plugin startup](https://github.com/waseem18/mattermost-plugin-google-calendar/blob/master/server/configuration.go#L34) on this param; this change prevents the plugin from installing and lead to :

```

  Must have secret key

  unable to start plugin: mattermost-google-calendar

  github.com/mattermost/mattermost-server/plugin.(*Environment).Activate

    /go/src/github.com/mattermost/mattermost-server/plugin/environment.go:216

  github.com/mattermost/mattermost-server/app.(*App).SyncPluginsActiveState

    /go/src/github.com/mattermost/mattermost-server/app/plugin.go:95

  github.com/mattermost/mattermost-server/app.(*App).InitPlugins.func2

    /go/src/github.com/mattermost/mattermost-server/app/plugin.go:172

  github.com/mattermost/mattermost-server/config.(*emitter).invokeConfigListeners.func1

    /go/src/github.com/mattermost/mattermost-server/config/emitter.go:35

  sync.(*Map).Range

    /usr/local/go/src/sync/map.go:333

  github.com/mattermost/mattermost-server/config.(*emitter).invokeConfigListeners

    /go/src/github.com/mattermost/mattermost-server/config/emitter.go:33

  github.com/mattermost/mattermost-server/config.(*commonStore).set

    /go/src/github.com/mattermost/mattermost-server/config/common.go:80

  github.com/mattermost/mattermost-server/config.(*FileStore).Set

    /go/src/github.com/mattermost/mattermost-server/config/file.go:97

  github.com/mattermost/mattermost-server/app.(*Server).UpdateConfig

    /go/src/github.com/mattermost/mattermost-server/app/config.go:53

  github.com/mattermost/mattermost-server/app.(*App).UpdateConfig

    /go/src/github.com/mattermost/mattermost-server/app/config.go:59

  github.com/mattermost/mattermost-server/app.(*App).EnablePlugin

    /go/src/github.com/mattermost/mattermost-server/app/plugin.go:245

  github.com/mattermost/mattermost-server/api4.enablePlugin

    /go/src/github.com/mattermost/mattermost-server/api4/plugin.go:187

  github.com/mattermost/mattermost-server/web.Handler.ServeHTTP

    /go/src/github.com/mattermost/mattermost-server/web/handlers.go:167

  github.com/mattermost/mattermost-server/vendor/github.com/gorilla/mux.(*Router).ServeHTTP

    /go/src/github.com/mattermost/mattermost-server/vendor/github.com/gorilla/mux/mux.go:162

  net/http.serverHandler.ServeHTTP

    /usr/local/go/src/net/http/server.go:2774

  net/http.(*conn).serve

    /usr/local/go/src/net/http/server.go:1878

  runtime.goexit

    /usr/local/go/src/runtime/asm_amd64.s:1337"
}
```

**Comment by unwmun-smobile** [comment]
I have the same problem.

**Comment by cinco** [comment]
> Looks like this setting has been removed by plugin.json [here](https://github.com/waseem18/mattermost-plugin-google-calendar/commit/c097f2c123be3ac2bc9665c6ea71795ac0ecb545#diff-b3568fa08e53a00f66841ed9ca2c53d4)
> 
> But, on master branch, there is still [a check on plugin startup](https://github.com/waseem18/mattermost-plugin-google-calendar/blob/master/server/configuration.go#L34) on this param; this change prevents the plugin from installing and lead to :
> 
> ```
> 
>   Must have secret key
> 
>   unable to start plugin: mattermost-google-calendar
> 
>   github.com/mattermost/mattermost-server/plugin.(*Environment).Activate
> 
>     /go/src/github.com/mattermost/mattermost-server/plugin/environment.go:216
> 
>   github.com/mattermost/mattermost-server/app.(*App).SyncPluginsActiveState
> 
>     /go/src/github.com/mattermost/mattermost-server/app/plugin.go:95
> 
>   github.com/mattermost/mattermost-server/app.(*App).InitPlugins.func2
> 
>     /go/src/github.com/mattermost/mattermost-server/app/plugin.go:172
> 
>   github.com/mattermost/mattermost-server/config.(*emitter).invokeConfigListeners.func1
> 
>     /go/src/github.com/mattermost/mattermost-server/config/emitter.go:35
> 
>   sync.(*Map).Range
> 
>     /usr/local/go/src/sync/map.go:333
> 
>   github.com/mattermost/mattermost-server/config.(*emitter).invokeConfigListeners
> 
>     /go/src/github.com/mattermost/mattermost-server/config/emitter.go:33
> 
>   github.com/mattermost/mattermost-server/config.(*commonStore).set
> 
>     /go/src/github.com/mattermost/mattermost-server/config/common.go:80
> 
>   github.com/mattermost/mattermost-server/config.(*FileStore).Set
> 
>     /go/src/github.com/mattermost/mattermost-server/config/file.go:97
> 
>   github.com/mattermost/mattermost-server/app.(*Server).UpdateConfig
> 
>     /go/src/github.com/mattermost/mattermost-server/app/config.go:53
> 
>   github.com/mattermost/mattermost-server/app.(*App).UpdateConfig
> 
>     /go/src/github.com/mattermost/mattermost-server/app/config.go:59
> 
>   github.com/mattermost/mattermost-server/app.(*App).EnablePlugin
> 
>     /go/src/github.com/mattermost/mattermost-server/app/plugin.go:245
> 
>   github.com/mattermost/mattermost-server/api4.enablePlugin
> 
>     /go/src/github.com/mattermost/mattermost-server/api4/plugin.go:187
> 
>   github.com/mattermost/mattermost-server/web.Handler.ServeHTTP
> 
>     /go/src/github.com/mattermost/mattermost-server/web/handlers.go:167
> 
>   github.com/mattermost/mattermost-server/vendor/github.com/gorilla/mux.(*Router).ServeHTTP
> 
>     /go/src/github.com/mattermost/mattermost-server/vendor/github.com/gorilla/mux/mux.go:162
> 
>   net/http.serverHandler.ServeHTTP
> 
>     /usr/local/go/src/net/http/server.go:2774
> 
>   net/http.(*conn).serve
> 
>     /usr/local/go/src/net/http/server.go:1878
> 
>   runtime.goexit
> 
>     /usr/local/go/src/runtime/asm_amd64.s:1337"
> }
> ```

Did you ever compile and test?

**Comment by ybudimirov** [comment]
@revolunet  
Were you able to finish setup of this plugin?

**Comment by revolunet** [comment]
yes, setup finished ok but this plugin had other issues (cant remember) so we dont use it :/

**Comment by ybudimirov** [comment]
thank you for reply!
